include/vinecopulib/bicop/implementation/class.ipp

```diff
@@ -560,20 +569,15 @@ Bicop::set_parameters(const Eigen::MatrixXd& parameters)
 
 //! @brief sets variable types.
 //! @param var_types a vector of size two specifying the types of the variables,
-//!   e.g., `{"c", "d"}` means first varible continuous, second discrete.
+//!   e.g., `{"c", "d"}` means first variable continuous, second discrete.
 inline void
 Bicop::set_var_types(const std::vector<std::string>& var_types)
 {
-  if (var_types.size() != 2) {
-    throw std::runtime_error("var_types must have size two.");
-  }
-  for (auto t : var_types) {
-    if (!tools_stl::is_member(t, { "c", "d" })) {
-      throw std::runtime_error("var type must be either 'c' or 'd'.");
-    }
-  }
+  check_var_types(var_types);
   var_types_ = var_types;
-  bicop_->set_var_types(var_types);
+  if (bicop_) {
+    bicop_->set_var_types(var_types);
+  }
```

Comment:
I think the `flip_var_types()` below also needs to be in this block as it modifies the `bicop_` object-